### ISSUE TITLE
Meta-Issue: Performance Testing

### ISSUE THREAD

**Issue by dannylamb** [Islandora/documentation#936]
This is a meta-issue to track the progress of performance testing the CLAW stack.  Please refer to this issue in any subsequent issues to link them.

Islandora CLAW's architecture was designed to be scalable, but we have yet to set up a complex distributed install and see how it performs under heavy load.  Statistics on performance can be tricky/slippery/misleading, so we'll want to isolate as many vectors as possible.  Things to consider are

- Performance between various installation setups (all-in-one vs. separate solr and fedora vs. load balanced microservices, etc..)
- Performance handling high volumes of objects (ingest 1000 vs 10000 vs 1000000 objects)
- Performance handling large files (what does a 1GB file do.  1TB?)
- Performance degradation as # of fields increases (added as per @mjordan and @DiegoPino's comments)

**Comment by mjordan** [comment]
Should we also be load testing Drupal itself for number of nodes and and number of requests? Might be useful to have a baseline while testing the Islandora-specific components/configurations.

**Comment by DiegoPino** [comment]
I agree Mark, IMHO the key is testing number of Nodes and Media entities based on Number of Content types, Media Types and attached Fields that Islandora provides/suggests, including taxonomy. Each Field is one DB table, so that piece matters

**Comment by dannylamb** [comment]
@mjordan @DiegoPino In some sense that would be tested with the "high number of objects" test, but testing # of fields is definitely a factor we give careful consideration.

**Comment by dannylamb** [comment]
Edited the issue description to include that.  Thanks guys.

**Comment by Natkeeran** [comment]
In theory, we should be able to deploy to multiple machines just with modifying Ansible configurations.  We need to this out as well.

**Comment by kstapelfeldt** [comment]
Related Issue: https://github.com/Islandora/documentation/issues/30
